src/components/shared/Terminal.tsx

```diff
@@ -104,15 +113,131 @@ function Terminal(prop: {
     return executeTouch(path, true);
   }
 
+  function executeRm(path: string, flags: string, directory = false): string[] {
+    // Make a copy to avoid mutating the original (it's a state variable)
+    const fileSystemCopy = _.cloneDeep(prop.fileSystem);
+    const cwdCopy = fileSystemCopy.getFileSystemObjectFromPath(
+      prop.currentWorkingDirectory.path
+    );
+    let fileName = '';
+    let dir;
+
+    // If the path is absolute, find the directory to create the file in
+    // Otherwise, use the current working directory
+    if (path.includes('/')) {
+      const pathArr = path.split('/');
+      fileName = pathArr.pop() || '';
+      if (path.startsWith('/')) {
+        dir = fileSystemCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      } else {
+        dir = cwdCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      }
+    } else {
+      fileName = path;
+      dir = cwdCopy;
+    }
+
+    const file = dir?.getChild(fileName);
+    if (!file) {
+      return [`rm: '${path}': No such file or directory`];
+    }
+
+    if (
+      (directory && file.isEmpty()) ||
+      !file.isDirectory ||
+      flags.includes('r')
+    ) {
+      if (path === '/') {
+        // TODO: Figure out how to handle this
+        return ["What are you doing? You can't delete the root directory!"];
+      }
+      dir.removeFileSystemObject(fileName);
+    } else {
+      if (directory && !file.isEmpty()) {
+        return [`rmdir: '${path}': Directory not empty`];
+      }
+      return [`rm: '${path}': is a directory`];
+    }
+
+    prop.setFileSystem(fileSystemCopy);
+    prop.setCurrentWorkingDirectory(cwdCopy);
+    return [];
+  }
+
+  function executeRmdir(path: string, flags: string): string[] {
+    return executeRm(path, flags, true);
+  }
+
+  function executeCopy(
+    path: string,
+    destination: string,
+    flags: string
+  ): string[] {
+    // Make a copy to avoid mutating the original (it's a state variable)
+    const fileSystemCopy = _.cloneDeep(prop.fileSystem);
+    const cwdCopy = fileSystemCopy.getFileSystemObjectFromPath(
+      prop.currentWorkingDirectory.path
+    );
+    const file = path.includes('/')
+      ? prop.fileSystem.getFileSystemObjectFromPath(path)
+      : prop.currentWorkingDirectory.getFileSystemObjectFromPath(path);
+    let destDir;
+    let newFileName;
+
+    // If the path is absolute, find the directory to create the file in
+    // Otherwise, use the current working directory
+    if (destination.includes('/')) {
+      const pathArr = destination.split('/');
+      newFileName = pathArr.pop() || '';
+      if (path.startsWith('/')) {
+        destDir = fileSystemCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      } else {
+        destDir = cwdCopy.getFileSystemObjectFromPath(pathArr.join('/'));
+      }
+    } else {
+      newFileName = destination;
+      destDir = cwdCopy;
+    }
+
+    if (!file) {
```

Comment:
Refactor - upgrade this check to be `file === null`. It'd be better if we explicitly check the lack of file instead of just handling undefined